### PR TITLE
Feat youtube data

### PR DESCRIPTION
**Pull request by mixcas**
The work was done in `feat-youtube-data` which branched out from `youtube` which was branched from `master`. The idea is to work on branches from `youtube` and eventually integrate vimeo and youtube parsing into a single plugin.

This replaces all copy text "vimeo" to "video" and "youtube" where appropriate.
Add a field to put down an api key (theres one in trello #insearchoftruth)
Parses data from youtube video and add it to proper fields

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+    "name": "cas/globie-video-sucker",
```
Comment (varffs) [inline composer.json]: `cas/` could that be `igv/`?

File: globie-vimeo-sucker.php [hunk 2]
```diff
@@ -45,73 +48,75 @@ public function enqueue_scripts( $hook ){
     $saved_post_types = get_option( 'gvsucker_settings_post_types' );
 
     if( in_array( $post_type, $saved_post_types ) ) {
-      wp_register_script( 'globie-vimeo-sucker-script', plugins_url( '/globie-vimeo-sucker.js', __FILE__ ), array( 'jquery' ) );
+      wp_register_script( 'globie-video-sucker-get-video-id', plugins_url( '/globie-video-sucker.js', __FILE__ ), array( 'jquery' ) );
```
Comment (varffs) [inline globie-vimeo-sucker.php:51]: you have to rename the file too!

**Review by varffs — CHANGES_REQUESTED** [review]
I think you have missed commits somewhere. You need to rename the file and also rename the `var suckDataButton` ID value to make it work. Once that is done it seems to work fine. But I'm wondering if there are other things you forgot to commit?

File: globie-vimeo-sucker.php (hunk 2 above)
Comment (mixcas) [reply globie-vimeo-sucker.php:51]: LOL will fix

File: composer.json (hunk 1 above)
Comment (mixcas) [reply composer.json]: ah sorry :3

**Review by varffs — APPROVED** [review]

**Comment by varffs** [comment]
Seems to be working fine. I do get this odd error:

```
VM761:1 Uncaught SyntaxError: Unexpected end of JSON input
    at JSON.parse (<anonymous>)
    at XMLHttpRequest.xmlhttp.onreadystatechange (globie-video-sucker.js?ver=4.8.1:89)
```

But doesnt break anything so going to merge and keep eyes open for bugs

**Pull request merged**